<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>

<!-- mavros -->
<arg name="fcu_url" default="/dev/ttyUSB0:921600" />
<arg name="gcs_url" default="" />
<arg name="tgt_system" default="1" />
<arg name="tgt_component" default="1" />
<arg name="log_output" default="screen" />
<arg name="fcu_protocol" default="v2.0" />
<arg name="respawn_mavros" default="false" />
<arg name="rosbag" default="false" />
<arg name="rosbag_image" default="false" />

<include file="$(find mavros)/launch/node.launch">
	<arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
	<arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />

	<arg name="fcu_url" value="$(arg fcu_url)" />
	<arg name="gcs_url" value="$(arg gcs_url)" />
	<arg name="tgt_system" value="$(arg tgt_system)" />
	<arg name="tgt_component" value="$(arg tgt_component)" />
	<arg name="log_output" value="$(arg log_output)" />
	<arg name="fcu_protocol" value="$(arg fcu_protocol)" />
	<arg name="respawn_mavros" default="$(arg respawn_mavros)" />
</include>


<!-- mavros_bridge -->
<arg name="lidar_x" value="0" />
<arg name="lidar_y" value="0" />
<arg name="lidar_z" value="0.1" />
<arg name="lidar_roll" value="0" />
<arg name="lidar_pitch" value="0" />
<arg name="lidar_yaw" value="1.5708" />

<param name="lidar_x" type="double" value="$(arg lidar_x)" />
<param name="lidar_y" type="double" value="$(arg lidar_y)" />
<param name="lidar_z" type="double" value="$(arg lidar_z)" />
<param name="lidar_roll" type="double" value="$(arg lidar_roll)" />
<param name="lidar_pitch" type="double" value="$(arg lidar_pitch)" />
<param name="lidar_yaw" type="double" value="$(arg lidar_yaw)" />

<node name="bridge_ros_node" pkg="lidar2px4" type="bridge_ros_node" output="screen"/>

</launch>
